File: es6/prod/src/platform/worker_app.js
import { Parse5DomAdapter } from 'angular2/src/platform/server/parse5_adapter';
import { PostMessageBus, PostMessageBusSink, PostMessageBusSource } from 'angular2/src/web_workers/shared/post_message_bus';
import { genericWorkerAppProviders } from './worker_app_common';
var _postMessage = postMessage;
export function setupWebWorker(zone) {
    Parse5DomAdapter.makeCurrent();
    var sink = new PostMessageBusSink({
        postMessage: (message, transferrables) => { _postMessage(message, transferrables); }
    });
    var source = new PostMessageBusSource();
    var bus = new PostMessageBus(sink, source);
    return genericWorkerAppProviders(bus, zone);
}

File: modules/angular2/src/platform/worker_app.dart
library angular2.src.platform.worker_app;

import 'package:angular2/src/core/zone/ng_zone.dart';
import 'package:angular2/src/platform/server/webworker_adapter.dart';
import 'package:angular2/src/platform/worker_app_common.dart';
import 'package:angular2/core.dart';
import 'package:angular2/src/web_workers/shared/isolate_message_bus.dart';
import 'package:angular2/src/web_workers/shared/message_bus.dart';
import 'dart:isolate';

const OpaqueToken RENDER_SEND_PORT = const OpaqueToken("RenderSendPort");

const List<dynamic> WORKER_APP_APPLICATION = const [
  WORKER_APP_APPLICATION_COMMON,
  const Provider(MessageBus,
      useFactory: createMessageBus, deps: const [NgZone, RENDER_SEND_PORT]),
  const Provider(APP_INITIALIZER, useValue: setupIsolate, multi: true)
];

MessageBus createMessageBus(NgZone zone, SendPort replyTo) {
  ReceivePort rPort = new ReceivePort();
  var sink = new WebWorkerMessageBusSink(replyTo, rPort);
  var source = new IsolateMessageBusSource(rPort);
  var bus = new IsolateMessageBus(sink, source);
  bus.attachToZone(zone);
  return bus;
}

setupIsolate() {
  WebWorkerDomAdapter.makeCurrent();
}

class WebWorkerMessageBusSink extends IsolateMessageBusSink {
  WebWorkerMessageBusSink(SendPort sPort, ReceivePort rPort) : super(sPort) {
    sPort.send(rPort.sendPort);
  }
}
